cmd/pgbackrest/main.go

```diff
@@ -240,3 +168,180 @@ func NewForConfig(config *rest.Config) (*KubeAPI, error) {
+
+	// parse the environment variable and store the appropriate boolean value
+	// we will discard the error and treat the value as "false" if it is not
+	// explicitly set
+	cfg.s3VerifyTLS, _ = strconv.ParseBool(os.Getenv("PGHA_PGBACKREST_S3_VERIFY_TLS"))
+	log.Debugf("PGHA_PGBACKREST_S3_VERIFY_TLS set to: %t", cfg.s3VerifyTLS)
+
+	return cfg
+}
+
+// createPGBackRestCommand form the proper pgBackRest command based on the configuration provided
+func createPGBackRestCommand(cfg config) []string {
+	cmd := []string{backrestCommand}
+
+	switch cfg.command {
+	default:
+		log.Fatalf("unsupported backup command specified: %s", cfg.command)
+	case pgtaskBackrestStanzaCreate:
+		log.Info("backrest stanza-create command requested")
+		cmd = append(cmd, backrestStanzaCreateCommand, cfg.commandOpts)
+	case pgtaskBackrestInfo:
+		log.Info("backrest info command requested")
+		cmd = append(cmd, backrestInfoCommand, cfg.commandOpts)
+	case pgtaskBackrestBackup:
+		log.Info("backrest backup command requested")
+		cmd = append(cmd, backrestBackupCommand, cfg.commandOpts)
+	}
+
+	if cfg.localS3Storage {
+		// if the first backup fails, still attempt the 2nd one
+		cmd = append(cmd, ";")
+		cmd = append(cmd, cmd...)
+		cmd[len(cmd)-1] = repoTypeFlagS3 // a trick to overwite the second ";"
+		// pass in the flag to disable TLS verification, if set
+		// otherwise, maintain default behavior and verify TLS
+		if !cfg.s3VerifyTLS {
+			cmd = append(cmd, noRepoS3VerifyTLS)
+		}
+		log.Info("backrest command will be executed for both local and s3 storage")
+	} else if cfg.repoType == "s3" {
+		cmd = append(cmd, repoTypeFlagS3)
+		// pass in the flag to disable TLS verification, if set
+		// otherwise, maintain default behavior and verify TLS
+		if !cfg.s3VerifyTLS {
+			cmd = append(cmd, noRepoS3VerifyTLS)
+		}
+		log.Info("s3 flag enabled for backrest command")
+	}
+
+	if cfg.localGCSStorage {
+		// if the first backup fails, still attempt the 2nd one
+		cmd = append(cmd, ";")
+		cmd = append(cmd, cmd...)
+		cmd[len(cmd)-1] = repoTypeFlagGCS // a trick to overwite the second ";"
+		log.Info("backrest command will be executed for both local and gcs storage")
+	} else if cfg.repoType == "gcs" {
+		cmd = append(cmd, repoTypeFlagGCS)
+		log.Info("gcs flag enabled for backrest command")
+	}
+
+	return cmd
+}
+
+// compareHashAndRunCommand calculates the hash of the pgBackRest configuration locally against
+// a hash of the pgBackRest configuration for the container being exec'd into to run a pgBackRest
+// command.  Only if the hashes match will the pgBackRest command be run, otherwise and error will
+// be written and exit code 1 will be returned.  This is done to ensure a pgBackRest command is only
+// run when it can be verified that the exepected configuration is present.
+func compareHashAndRunCommand(kubeapi *KubeAPI, cfg config, cmd []string) (string, string, error) {
+	const hashCmd = "sha1sum /etc/pgbackrest/conf.d/* | sha1sum"
```

Comment:
🔧 This differs from the remote hashing. Can `hashCmd` become the whole globstar + sum and be used by both?

```go
	const hashCmd = `
shopt -s globstar
…
sha1sum "${files[@]}" | sha1sum
`
```

then

```go
	script := `
declare -r hash="$1"
local_hash="$(` + hashCmd + `)"
…
`
```

Reply:
Yea, they definitely can.  Both now share the same base script for grabbing the file.  Thanks!